pkg/storage/storage.go

```diff
@@ -181,3 +184,48 @@ func parseCommit(r io.Reader, hash string) (Commit, error) {
 
 	return c, nil
 }
+
+//TreeEntry is a file or folder at a given path in a repository
+type TreeEntry struct {
+	Mode   string
+	Type   string
+	Object string
+	Path   string
+}
+
+//Tree returns the files and folders at a given rev at a path in a repository
+func (s *storage) Tree(ctx context.Context, owner string, name string, rev string, path string) ([]TreeEntry, error) {
+	args := []string{"ls-tree", rev, path}
```

Comment:
I would use `ls-tree -z` for easier parsing

Reply:
I would also pass in `--long` to get the blob-size :thinking: 

> -l
> --long
> 
> Show object size of blob (file) entries.

https://git-scm.com/docs/git-ls-tree#git-ls-tree--l

Reply:
I won't be using `-z` as it doesn't add that much, I think. Parsing is just easier with a linebreak and shouldn't differ in many characters.
Thanks for the hint on `-l`. I saw that too after looking into `-z`, but don't see a use case for having it in the API as we speak. We won't show the size in the file browser, just not that important.
In case this changes in the future, we can still add `size` to the API without breaking anything. :blush: